### DISCUSSION TITLE
Create Badge with Custom Icon

### DISCUSSION THREAD

**Discussion by transcental** [badges/shields#6999] [Q&A] [answered]
Hi, 

I believe that it is possible to create badges with custom icons, but, I'm not sure how.
I want to create a badge for [Replit](https://replit.com), with this logo: ![https://upload.wikimedia.org/wikipedia/commons/thumb/b/b2/Repl.it_logo.svg/1024px-Repl.it_logo.svg.png](https://upload.wikimedia.org/wikipedia/commons/thumb/b/b2/Repl.it_logo.svg/1024px-Repl.it_logo.svg.png).

Please could someone tell me how to do this? *And, how can I apply the `?style=for-the-badge` to this?*

**Comment by calebcartwright** [comment, marked as answer]
The documentation for adding logos to your badge, including custom logos, can be found both on the [Shields.io site under the styles section](https://shields.io/#styles), and in our [logo documentation](https://github.com/badges/shields/blob/master/doc/logos.md#custom-logos), and in various past Discussions where this question has previously been asked and answered, e.g. #6882


<html><body>


?logo=appveyor | Insert one of the named logos from (bitcoin, dependabot, gitlab, npm, paypal, serverfault, stackexchange, superuser, telegram, travis) or simple-icons. Simple-icons are referenced using icon slugs which can be found on the simple-icons site or in the slugs.md file in the simple-icons repository.
-- | --
?logo=data:image/png;base64,… | Insert  custom logo image (≥ 14px high). There is a limit on the total size of  request headers we can accept (8192 bytes). From a practical  perspective, this means the base64-encoded image text is limited to  somewhere slightly under 8192 bytes depending on the rest of the request  header.
?logoColor=violet | Set  the color of the logo (hex, rgb, rgba, hsl, hsla and css named colors  supported). Supported for named logos but not for custom logos.
?logoWidth=40 | Set the horizontal space to give to the logo


</body>
</html>

**Reply by transcental** [reply]
Thank you!
